### PR TITLE
[Task]: Update Optimize Input Designs

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Description

- [x] Updating Optimize's input UI as per figma.
- [x] Update state to have constraintGroups - will be utilized for UI now, and API later.
- [x] Add a new constraint form (to be manipulated similar to policy interventions)
- [x] a bunch of rewriting text + minor styling updates

Note: 
Trying to keep minimal structural changes to the optimizePayload as these will be completed along side the pyciemss-service update in a follow up PR
Therefore there should be no functional difference in the before + after for the API call, results, ect.
All of the constraints past the first are currently ignored for this reason

The exact wording for all of these boxes is being updated today and potentially tomorrow but to keep moving and to reduce PR sizes I will update these in a later PR

# Rough screenshot:
image

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-constraint-group-form.vue [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+				optionLabel="label"
+				optionValue="value"
+				:options="[
+					{ label: 'less than', value: true },
+					{ label: 'greater than', value: false }
+				]"
+				@update:model-value="emit('update-self', config)"
+			/>
+			<p>a threshold of</p>
+			<InputNumber
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-constraint-group-form.vue:63]: Do we want to use this or the `tera-input-number.vue` @blanchco made?

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-constraint-group-form.vue [hunk 2]
```diff
@@ -0,0 +1,148 @@
+<template>
+	<div class="container">
+		<div class="form-header">
+			<div>
+				<InputText
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-constraint-group-form.vue:5]: What about `tera-input.vue`?

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-constraint-group-form.vue [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	isEditing.value = !isEditing.value;
+};
+</script>
+
+<style scoped>
+.container {
+	width: 100%;
+	display: flex;
+	margin-top: 1rem;
+	padding: 1rem 1rem 1rem 1.5rem;
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-constraint-group-form.vue:104]: use gap variables

**Review by shawnyama — APPROVED** [review]

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-constraint-group-form.vue (hunk 1 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-constraint-group-form.vue]: Will use tera-input
Thanks

File: packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-constraint-group-form.vue (hunk 2 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/workflow/ops/optimize-ciemss/tera-optimize-constraint-group-form.vue]: Thanks

**Review by YohannParis — APPROVED** [review]

File: packages/client/hmi-client/src/components/widgets/tera-input.vue [hunk 4]
```diff
@@ -4,6 +4,7 @@
 		<main :class="{ error: getErrorMessage }" @click.self.stop="focusInput">
 			<input
 				ref="inputField"
+				v-bind="$attrs"
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/widgets/tera-input.vue:7]: what is this change for?

Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/widgets/tera-input.vue:7]: This is so I can pass things such as "disabled" to this component
This was @shawnyama's change

**Pull request merged**